cli/types.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+type OptionTypeConstructor<T> = T extends boolean ? BooleanConstructor
+  : T extends number ? NumberConstructor
+  : T extends string ? StringConstructor
+  : BooleanConstructor | NumberConstructor | StringConstructor;
+
+export interface Option<T> {
+  name: string;
+  alias?: string;
+  description?: string;
+  default?: T;
+  type?: OptionTypeConstructor<T>;
+  negatable?: boolean;
+  value?: {
+    name: string;
+    optional?: boolean;
+    multiple?: boolean;
+    requireEquals?: boolean;
+  };
```

Comment:
I don't think this nesting is necessary and the value's name field can be removed (does it even do anything?)

Reply:
See below.